### DISCUSSION TITLE
How to handle dependent attributes in pimcore

### DISCUSSION THREAD

**Discussion by KasthuriAnandhan** [pimcore/pimcore#16792] [Q&A / Support]
Hello everyone,

When creating a product and selecting a brand, all available categories are displayed in the category attribute. However, I need to restrict the displayed categories to only those associated with the selected brand. Similarly, I'd like subcategories to be filtered based on the parent category selection.

Is it possible to achieve this functionality within Pimcore? If so, could someone please provide guidance on how to implement it?

Any help or insights on this matter would be greatly appreciated.

Thank you.

**Comment by KasthuriAnandhan** [comment]
Anybody can assist us here?

**Reply by AlephLau** [reply]
@KasthuriAnandhan  As far as I'm aware, there isn't a built-in way to filter a field, such as the brand in this case, based on another field, such as the category, natively within Pimcore. You'll likely need to implement this functionality yourself.

One approach could involve creating custom JavaScript logic to dynamically update the available options for the brand field based on the selected category. This would involve listening for changes to the category field and then fetching and updating the options for the brand field accordingly.

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90  Do you have any idea about it.

**Reply by kingjia90** [reply]
Unfortunately i had not yet the time to take a look to it, it was quite busy this week due to the Pimcore Inspire 2024. 
I also noted that you wrote this to the Enterprise support as well, so i just wanted to reassure that it would be taken care ASAP and have acknowledged the assistance request

**Comment by KasthuriAnandhan** [comment]
Okay @kingjia90 , Thank you.

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90 Could you please confirm if any progress has been made on my request?

**Reply by kingjia90** [reply]
Where are these category configured and set? in a multiselect field or using the relational? For select field, maybe with the [dynamic select type](https://pimcore.com/docs/platform/2023.1/Pimcore/Objects/Object_Classes/Data_Types/Dynamic_Select_Types/) by checking the `context['object']` but afaik is not in updating on select/change event but on reload, out-of-the box there are no tools for what you are trying to achieve.

**Comment by KasthuriAnandhan** [comment]
Brand, Category, Sub Category are created as a select data type in class.
![image](https://github.com/pimcore/pimcore/assets/125338676/946c9d71-461a-42a0-97ee-064fc2ad6200)
Refer the image, if I selected the brand name in Brand field the respective categories of brand only will have to display in Category field.

**Reply by kingjia90** [reply]
Mhh, i see, so far there are no any solution coming to my mind other than making custom ajax calls or mapping the fields somewhere in JS, hope somebody from the community could provide their experience on this

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90 mapping the fields meant what?, and I have relational data type too for Brand, Category and Sub Category to the products. Is any possibility to achieve my request with this option.

**Reply by kingjia90** [reply]
was thinking something like, in js, trigger something on change that reads the relation values and hide the options that are not related to the selected value of another field, but again, no any prior experience on that

**Comment by KasthuriAnandhan** [comment]
Is anybody other have an idea about it?

**Comment by BigJabber** [comment]
If I'm understanding correctly, we do some similar dropdown filtering. To achieve what you've detailed above I'd be filling the select options using an options provider class. It isn't a perfect solution as it requires saves and reloads to be made to refresh the options but it works for us. A client side request as mentioned by @kingjia90 would be the optimal way to go probably but we've never had a reason to implement it.

https://pimcore.com/docs/platform/2023.2/Pimcore/Objects/Object_Classes/Data_Types/Dynamic_Select_Types/

**Comment by KasthuriAnandhan** [comment]
My PIMCORE version is v10.5.21, to use which is you said should I have to update my PIMCORE version?

**Reply by BigJabber** [reply]
No, I think this has been available for some time. Here's the documentation for 10.5 
https://pimcore.com/docs/pimcore/10.5/Development_Documentation/Objects/Object_Classes/Data_Types/Dynamic_Select_Types.html

**Comment by KasthuriAnandhan** [comment]
Should I write a custom code to achieve this @BigJabber.
